### ISSUE TITLE
Smarter early stopping

### ISSUE THREAD

**Issue by mspruc** [Mylos97/Thesis-ML#1]
Currently the patience set for early stopping is always 10, either we need to move over to a function so we do not miss out on global optima in regards to validation loss, or include it as part of the hyperparameter bayesian optimisation.

see paper 'Early Stopping | but when?'
